src/services/LibraryService.ts

```diff
@@ -89,4 +97,36 @@ export class LibraryService implements Service<fhir4.Library> {
 
     return createLibraryPackageBundle(library[0], req.query['include-terminology'] ?? false);
   }
+
+  /**
+   * result of sending a POST request to:
+   * {BASE_URL}/4_0_1/Library/$submit or {BASE_URL}/4_0_1/Library/:id/$submit
+   * POSTs a new artifact in "draft" status. The operation results in an error if the artifact
+   * does not have status set to "draft."
+   */
+  async submit(args: RequestArgs, { req }: RequestCtx) {
+    logger.info(`${req.method} ${req.path}`);
+
+    const contentType: string | undefined = req.headers['content-type'];
+    checkContentTypeHeader(contentType);
+
+    const resource = req.body;
+    checkExpectedResourceType(resource.resourceType, 'Library');
+
+    // check for "draft" status on the resource
+    if (resource.status !== 'draft') {
+      throw new BadRequestError(`The artifact must be in 'draft' status.`);
+    }
```

Comment:
Probably very unnecessary, but I think this could be added to `validationUtils.ts` if you wanted!

Reply:
I'll hold of on that for now since this is just a simple if statement, but if we end up checking draft status across several other operations then we can consider moving this into a helper